python/src/trezorlib/_internal/emulator.py

```diff
@@ -180,17 +185,17 @@ def stop(self):
         _rm_f(self.profile_dir / "trezor.port")
         self.process = None
 
-    def restart(self):
+    def restart(self) -> None:
         self.stop()
         self.start()
 
-    def __enter__(self):
+    def __enter__(self) -> "Emulator":
         return self
 
-    def __exit__(self, exc_type, exc_value, traceback):
+    def __exit__(self, exc_type, exc_value, traceback) -> None:
```

Comment:
there are missing type arguments, why does this not cause errors?

Reply:
`Pyright` does not check this by default (same as `mypy`). We would need to add `"reportMissingParameterType": true` into the config, but it seems too strict to me, there is a lot of "strange" functions that I could not find the argument type

Reply:
istm `reportMissingParameterType` is something we really want. please try to enable it

Reply:
Alright, got 48 errors of missing type annotation - so I will replace them with `Any` when I would really not know the real type

Reply:
Nice, it uncovered even some forgotten functions that could be reliably typed.

What is strange, I was not able to find the equivalent for return values in pyright docs - https://github.com/microsoft/pyright/blob/main/docs/configuration.md

In `strict` mode, the `Return type is unknown` errors are labeled as `reportUnknownVariableType` - but enabling this would mean a lot of extra "errors"

Reply:
`reportMissingParameterType` implemented and enforced in 69044dd